extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/authorization/db/cache/CoordinatorPollingBasicAuthorizerCacheManager.java

```diff
@@ -0,0 +1,291 @@
+    this.cachedRoleMaps = new ConcurrentHashMap<>();
+    this.authorizerPrefixes = new HashSet<>();
+    this.druidLeaderClient = druidLeaderClient;
+  }
+
+  @LifecycleStart
+  public void start()
+  {
+    if (!lifecycleLock.canStart()) {
+      throw new ISE("can't start.");
+    }
+
+    LOG.info("Starting CoordinatorPollingBasicAuthorizerCacheManager.");
+
+    try {
+      initUserMaps();
+
+      ScheduledExecutors.scheduleWithFixedDelay(
+          exec,
+          new Duration(commonCacheConfig.getPollingPeriod()),
+          new Duration(commonCacheConfig.getPollingPeriod()),
+          () -> {
+            try {
+              long randomDelay = ThreadLocalRandom.current().nextLong(0, commonCacheConfig.getMaxRandomDelay());
+              LOG.debug("Inserting random polling delay of [%s] ms", randomDelay);
+              Thread.sleep(randomDelay);
+
+              LOG.debug("Scheduled cache poll is running");
+              for (String authorizerPrefix : authorizerPrefixes) {
+                UserAndRoleMap userAndRoleMap = fetchUserAndRoleMapFromCoordinator(authorizerPrefix, false);
+                if (userAndRoleMap != null) {
+                  cachedUserMaps.put(authorizerPrefix, userAndRoleMap.getUserMap());
+                  cachedRoleMaps.put(authorizerPrefix, userAndRoleMap.getRoleMap());
+                }
+              }
+              LOG.debug("Scheduled cache poll is done");
+            }
+            catch (Throwable t) {
+              LOG.makeAlert(t, "Error occured while polling for cachedUserMaps.").emit();
+            }
+          }
+      );
+
+      lifecycleLock.started();
+      LOG.info("Started CoordinatorPollingBasicAuthorizerCacheManager.");
+    }
+    finally {
+      lifecycleLock.exitStart();
+    }
+  }
+
+  @LifecycleStop
+  public void stop()
+  {
+    if (!lifecycleLock.canStop()) {
+      throw new ISE("can't stop.");
+    }
+
+    LOG.info("CoordinatorPollingBasicAuthorizerCacheManager is stopping.");
+    exec.shutdown();
+    LOG.info("CoordinatorPollingBasicAuthorizerCacheManager is stopped.");
+  }
+
+  @Override
+  public void handleAuthorizerUpdate(String authorizerPrefix, byte[] serializedUserAndRoleMap)
+  {
+    LOG.debug("Received cache update for authorizer [%s].", authorizerPrefix);
+    Preconditions.checkState(lifecycleLock.awaitStarted(1, TimeUnit.MILLISECONDS));
+    try {
+      UserAndRoleMap userAndRoleMap = objectMapper.readValue(
+          serializedUserAndRoleMap,
+          BasicAuthUtils.AUTHORIZER_USER_AND_ROLE_MAP_TYPE_REFERENCE
+      );
+
+      cachedUserMaps.put(authorizerPrefix, userAndRoleMap.getUserMap());
+      cachedRoleMaps.put(authorizerPrefix, userAndRoleMap.getRoleMap());
+
+      if (commonCacheConfig.getCacheDirectory() != null) {
+        writeMapToDisk(authorizerPrefix, serializedUserAndRoleMap);
+      }
+    }
+    catch (Exception e) {
+      LOG.makeAlert(e, "WTF? Could not deserialize user/role map received from coordinator.").emit();
```

Comment:
Is this intended to does nothing for all kinds of exceptions but emitting an alert?

Reply:
Yes, see the previous comment